modular_citadel/code/modules/festive/vectorcraft.dm

```diff
@@ -0,0 +1,581 @@
+				var/mob/living/carbon/C = i
+				driver = C
+				to_chat(driver, "<span class='notice'><b>You shuffle across to the driver's seat of the [src]</b></span>")
+				start_engine()
+				break
+		if(!driver)
+			stop_engine()
+			return
+	if(!gear)
+		gear = driver.a_intent
+	if(gear == "auto")
+		return
+	//USE THE CCLUUUTCHHH
+	if(gear != driver.a_intent && !driver.combatmode)
+		//playsound
+		to_chat(driver, "<span class='warning'><b>The gearbox gives out a horrific sound!</b></span>")
+		playsound(src.loc,'sound/vehicles/clutch_fail.ogg', 70, 0)
+		apply_damage(5)
+		acceleration = acceleration/2
+	else if(gear != driver.a_intent && driver.combatmode)
+		playsound(src.loc,'sound/vehicles/clutch_win.ogg', 100, 0)
+	gear = driver.a_intent
+
+//Bounce the car off a wall
+/obj/vehicle/sealed/vectorcraft/proc/bounce()
+	vector["x"] = -vector["x"]/2
+	vector["y"] = -vector["y"]/2
+	acceleration /= 2
+
+/obj/vehicle/sealed/vectorcraft/proc/ricochet(x_move, y_move)
+	var/speed = calc_speed()
+	apply_damage(speed/10)
+	bounce()
+
+//////////////////////////////////////////////////////////////
+//					Damage procs							//
+//////////////////////////////////////////////////////////////
+//Repairing
+/obj/vehicle/sealed/vectorcraft/attackby(obj/item/O, mob/user, params)
+	.=..()
+	if(istype(O, /obj/item/weldingtool) && user.a_intent != INTENT_HARM)
+		if(obj_integrity < max_integrity)
+			if(!O.tool_start_check(user, amount=0))
+				return
+
+			user.visible_message("[user] begins repairing [src].", \
+				"<span class='notice'>You begin repairing [src]...</span>", \
+				"<span class='italics'>You hear welding.</span>")
+
+			if(O.use_tool(src, user, 40, volume=50))
+				to_chat(user, "<span class='notice'>You repair [src].</span>")
+				apply_damage(-max_integrity)
+		else
+			to_chat(user, "<span class='notice'>[src] does not need repairs.</span>")
+
+//Heals/damages the car
+/obj/vehicle/sealed/vectorcraft/proc/apply_damage(damage)
+	obj_integrity -= damage
+	var/healthratio = ((obj_integrity/max_integrity)/2) + 0.5
+	max_acceleration = initial(max_acceleration) * healthratio
+	max_deceleration = initial(max_deceleration) * healthratio
+	boost_power = initial(boost_power) * healthratio
+
+	if(obj_integrity <= 0)
+		mob_exit(driver)
+		var/datum/effect_system/reagents_explosion/e = new()
+		var/turf/T = get_turf(src)
+		e.set_up(1, T, 1, 3)
+		e.start()
+		visible_message("The [src] explodes from taking too much damage!")
+		qdel(src)
+	if(obj_integrity > max_integrity)
+		obj_integrity = max_integrity
+
+//
+/obj/vehicle/sealed/vectorcraft/Bump(atom/M)
+	var/speed = calc_speed()
+	if(iscarbon(M))
+		var/mob/living/carbon/C = M
+		var/atom/throw_target = get_edge_target_turf(C, calc_angle())
+		C.throw_at(throw_target, 10, 14)
+		C.adjustBruteLoss(speed/10)
+		to_chat(C, "<span class='warning'><b>You are hit by the [src]!</b></span>")
+		to_chat(driver, "<span class='warning'><b>You just ran into [C] you crazy lunatic!</b></span>")
```

Comment:
And perhaps move the `throw_at` and `adjustBruteloss` lines below the chat messages since a frew mobs can end up being deleted from lethal damage.